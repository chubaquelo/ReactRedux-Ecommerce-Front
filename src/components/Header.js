import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';

const Header = () => (
  <Router>
    <header className="h-15percent-vh flex flex-row justify-center mt-3 w-10/12 mx-auto">
      <div id="logo" className="w-2/12 flex items-center justify-center">
        <Link to="/">
          <img
            src="../assets/logo-jewelry.png"
            alt="Logo"
            className="cursor-pointer"
            width="100"
          />
        </Link>
      </div>
      <div className="header-right-col w-10/12 flex flex-col">
        <div className="filita-de-arriba text-right py-1 text-black text-sm pr-3">
          <p className="italic text-gray-500 mr-2">Exclusive jewelry. Special for gifts.</p>
        </div>
        <div className="filita-de-abajo flex flex-row h-full">
          <nav
            id="navbar"
            className="w-12/12 md:w-10/12 flex items-center w-full px-6"
          >
            <ul className="flex flex-row w-full justify-around px-10">
              <Link to="/">
                <li className="cursor-pointer text-black font-bold">Home</li>
              </Link>
              <Link to="/">
                <li className="cursor-pointer text-black font-bold hidden sm:inline-flex">
                  Products
                </li>
              </Link>
              <Link to="/not-found">
                <li className="cursor-pointer text-black font-bold hidden sm:inline-flex">
                  Accesories
                </li>
              </Link>
              <Link to="/not-found">
                <li className="cursor-pointer text-black font-bold hidden sm:inline-flex">
                  Contact
                </li>
              </Link>
            </ul>
          </nav>
          <div
            id="cart-btn"
            className="flex items-center justify-center w-2/12 hidden lg:inline-flex"
          >
            <button
              type="button"
              className="py-1 bg-yellow-300 px-2 cursor-pointer hover:bg-yellow-400 text-black shadow-md"
            >
              Cart: 0 Items
            </button>
          </div>
        </div>
      </div>
    </header>
  </Router>
);

export default Header;
