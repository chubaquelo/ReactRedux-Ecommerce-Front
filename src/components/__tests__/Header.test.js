import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('renders correctly', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
