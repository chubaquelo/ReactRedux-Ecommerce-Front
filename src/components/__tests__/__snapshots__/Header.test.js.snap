// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders correctly 1`] = `
<header
  className="h-15percent-vh flex flex-row justify-center mt-3 w-10/12 mx-auto"
>
  <div
    className="w-2/12 flex items-center justify-center"
    id="logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        className="cursor-pointer"
        src="../assets/logo-jewelry.png"
        width="100"
      />
    </a>
  </div>
  <div
    className="header-right-col w-10/12 flex flex-col"
  >
    <div
      className="filita-de-arriba text-right py-1 text-black text-sm pr-3"
    >
      <p
        className="italic text-gray-500 mr-2"
      >
        Exclusive jewelry. Special for gifts.
      </p>
    </div>
    <div
      className="filita-de-abajo flex flex-row h-full"
    >
      <nav
        className="w-12/12 md:w-10/12 flex items-center w-full px-6"
        id="navbar"
      >
        <ul
          className="flex flex-row w-full justify-around px-10"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <li
              className="cursor-pointer text-black font-bold"
            >
              Home
            </li>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <li
              className="cursor-pointer text-black font-bold hidden sm:inline-flex"
            >
              Products
            </li>
          </a>
          <a
            href="/not-found"
            onClick={[Function]}
          >
            <li
              className="cursor-pointer text-black font-bold hidden sm:inline-flex"
            >
              Accesories
            </li>
          </a>
          <a
            href="/not-found"
            onClick={[Function]}
          >
            <li
              className="cursor-pointer text-black font-bold hidden sm:inline-flex"
            >
              Contact
            </li>
          </a>
        </ul>
      </nav>
      <div
        className="flex items-center justify-center w-2/12 hidden lg:inline-flex"
        id="cart-btn"
      >
        <button
          className="py-1 bg-yellow-300 px-2 cursor-pointer hover:bg-yellow-400 text-black shadow-md"
          type="button"
        >
          Cart: 0 Items
        </button>
      </div>
    </div>
  </div>
</header>
`;
